### ISSUE TITLE
linter: Add eslint/object-shorthand rule

### ISSUE THREAD

**Issue by Jkker** [oxc-project/oxc#17410]
## Summary

Request to implement the [`eslint/object-shorthand`](https://eslint.org/docs/latest/rules/object-shorthand) rule in oxlint.

## Rule Description

This rule enforces the use of shorthand syntax for object properties and methods. It's a commonly used stylistic rule that helps keep code concise.

### Examples

```javascript
// ❌ Incorrect (without shorthand)
const foo = {
  x: x,
  y: y,
  z: function() { return 1; }
};

// ✅ Correct (with shorthand)
const foo = {
  x,
  y,
  z() { return 1; }
};
```

## Use Case

This rule is part of the ESLint recommended stylistic rules and is commonly enabled in many projects. I'm migrating from ESLint to oxlint and noticed this rule is not yet implemented (tracked in #479 as TODO).

## Options

The ESLint rule supports several options:
- `"always"` (default) - requires shorthand syntax
- `"consistent"` - requires all properties to use consistent style
- `"consistent-as-needed"` - consistent, but prefer shorthand when possible
- `"methods"` - only require shorthand for methods
- `"properties"` - only require shorthand for properties
- `"never"` - never require shorthand

Reference: https://eslint.org/docs/latest/rules/object-shorthand

## Related

- Umbrella issue: #479

**Comment by camc314** [comment]
This is already being tracked by #479.
Please don't create issues for a specific rule unless:
  - you are willing to provide sponsorship for the issue (it's going to be clearer if this is a separate issue rather than being mixed in with every eslint issue)
  - the rule you would like to see implemented isn't covered by one of the existing issues

These extra issues adds an additional maintence burden on maintainers and makes it harder to track what's a real issue vs what's a duplicate.

Thanks for your understanding
